app/src/main/java/com/waseefakhtar/doseapp/feature/home/HomeScreen.kt

```diff
@@ -137,73 +179,95 @@ fun DailyMedications(state: HomeState, viewModel: HomeViewModel) {
         items(
             items = state.medications.sortedBy { it.medicationTaken },
             itemContent = {
-                MedicationCard(medication = it, viewModel)
+                MedicationCard(
+                    medication = it,
+                    viewModel = viewModel,
+                    showDeleteDialog = showDeleteDialog
+                )
             }
         )
     }
 }
 
+@OptIn(ExperimentalFoundationApi::class)
 @Composable
-fun MedicationCard(medication: Medication, viewModel: HomeViewModel) {
+fun MedicationCard(
+    showDeleteDialog: (medication : Medication) -> Unit,
```

Comment:
MedicationCard shouldn't known what this HOF will do. This parameter could be called onLongClick without passing medication through the parameter. It increase the component reusability.

Reply:
I've made the changes!